Dockerfile

```diff
@@ -1,22 +1,54 @@
-FROM python:2.7-slim
+FROM alpine:3.3
 
-WORKDIR /app
+ENV BASE_URL=False IMAGE_PROXY=False
 
-RUN useradd searx
+EXPOSE 8888
 
-EXPOSE 5000
-CMD ["/usr/local/bin/uwsgi", "--uid", "searx", "--gid", "searx", "--http", ":5000", "-w",  "searx.webapp"]
+WORKDIR /usr/local/searx
 
-RUN apt-get update && \
-    apt-get install -y --no-install-recommends \
-            zlib1g-dev libxml2-dev libxslt1-dev libffi-dev build-essential \
-            libssl-dev openssl && \
-    rm -rf /var/lib/apt/lists/*
+CMD ["./run.sh"]
 
-RUN pip install --no-cache uwsgi
+RUN adduser -D -h /usr/local/searx -s /bin/sh searx searx \
+ && echo '#!/bin/sh' >> run.sh \
+ && echo 'sed -i "s|base_url : False|base_url : $BASE_URL|g" searx/settings.yml' >> run.sh \
+ && echo 'sed -i "s/image_proxy : False/image_proxy : $IMAGE_PROXY/g" searx.setting.yml' >> run.sh \
+ && echo 'python searx/webapp.py' >> run.sh \
+ && chmod +x run.sh
 
-COPY requirements.txt /app/requirements.txt
-RUN pip install --no-cache -r requirements.txt
+COPY requirements.txt /usr/local/searx/requirements.txt
 
-COPY . /app
-RUN sed -i -e "s/ultrasecretkey/`openssl rand -hex 16`/g" searx/settings.yml
+RUN apk -U add \
+    build-base \
+    python \
+    python-dev \
+    py-pip \
+    libxml2 \
+    libxml2-dev \
+    libxslt \
+    libxslt-dev \
+    libffi-dev \
+    openssl \
+    openssl-dev \
+    ca-certificates \
+ && pip install --no-cache -r /usr/local/searx/requirements.txt \
+ && apk del \
+    build-base \
+    python-dev \
+    py-pip\
+    libffi-dev \
+    openssl-dev \
+    libxslt-dev \
+    libxml2-dev \
+    openssl-dev \
+    ca-certificates \
+ && rm -f /var/cache/apk/*
+
+COPY . /usr/local/searx
+
+RUN chown -R searx:searx /usr/local/searx
+
+USER searx
+
+RUN cd /usr/local/searx \
+ && sed -i "s/127.0.0.1/0.0.0.0/g" searx/settings.yml \
+ && sed -i "s/ultrasecretkey/`openssl rand -hex 16`/g" searx/settings.yml
```

Comment:
This is totally right, this is what I do that for my automated build : the key is generated the first time the container is started. It's not a problem here since the Dockerfile should be manually built : a unique secret key is generated during the build time. But I think like you, that moving this generation to run.sh brings more flexibility.